clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -929,6 +948,16 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
   if (getLangOpts().SYCLIsHost && D && D->hasAttr<SYCLKernelAttr>())
     Fn->addFnAttr("sycl_kernel");
 
+  if (getLangOpts().OpenCL || getLangOpts().SYCLIsHost) {
+    // Add metadata for a kernel function.
+    if (const FunctionDecl *FD = dyn_cast_or_null<FunctionDecl>(D)) {
+      EmitOpenCLKernelSubGroupMetadata(FD, Fn);
+
+      if (getLangOpts().SYCLIsDevice)
+        CGM.getSYCLRuntime().actOnFunctionStart(*FD, *Fn);
+    }
+  }
+
   if (getLangOpts().OpenCL || getLangOpts().SYCLIsDevice) {
```

Comment:
We need to keep the codegen part for intel_reqd_sub_group_size attribute on device here since test/CodeGenSYCL/sycl-multi-kernel-attr.cpp verifies multiple attributes such as  reqd_sub_group_size and intel_reqd_sub_group_size attribute together. So i am generating metadata for intel_reqd_sub_group_size attribute separately on host and device since reqd_sub_group_size  attribute is enabled on device only.

class Functor {
public:
  [[intel::reqd_sub_group_size(4), cl::reqd_work_group_size(32, 16, 16)]] void operator()() const {}
};